### PR TITLE
Fix build on Mac OS X

### PR DESCRIPTION
**Pull request by cajun-rat**

### CODE REVIEW

File: src/libaktualizr/asn1/asn1_message.cc [hunk 1]
```diff
@@ -29,7 +35,7 @@ int Asn1SocketWriteCallback(const void* buffer, size_t size, void* priv) {
 
   while (len > 0) {
     // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-pointer-arithmetic)
-    ssize_t written = send(*sock, b + pos, len, MSG_NOSIGNAL);
+    ssize_t written = send(*sock, b + pos, len, flags_no_signal);
```
Comment (jochenschneider) [inline src/libaktualizr/asn1/asn1_message.cc]: You can also ignore SIGPIPE like this. I don't know if that is needed.

    int val = 1;
    setsockopt(*sock, SOL_SOCKET, SO_NOSIGPIPE, (void*)&val, sizeof(val));

File: src/aktualizr_secondary/socket_server.cc [hunk 2]
```diff
@@ -7,7 +7,7 @@
 #include "utilities/dequeue_buffer.h"
 #include "utilities/sockaddr_io.h"
 
-#include <linux/tcp.h>
+#include <netinet/tcp.h>
```
Comment (jochenschneider) [inline src/aktualizr_secondary/socket_server.cc]: Does this work on Linux as well? That would be cool.

Comment (cajun-rat) [reply src/aktualizr_secondary/socket_server.cc]: It looks like it.  The CI failures were because of a clang-format problem, so hopefully this will pass.

**Comment by lbonn** [comment]
Superseeded by #843 (already merged)

**Pull request closed without merging**